tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -125,58 +128,111 @@ describe(__filename, () => {
     expect(root.find(UserProfileEditPicture)).toHaveLength(1);
   });
 
-  it('dispatches fetchUserAccount action if username is not found', () => {
+  it('dispatches fetchUserAccount and fetchUserNotifications actions if username is not found', () => {
     const { store } = signInUserWithUsername('tofumatt');
     const dispatchSpy = sinon.spy(store, 'dispatch');
 
     const username = 'i-am-not-tofumatt';
     const root = renderUserProfileEdit({ params: { username }, store });
 
+    sinon.assert.callCount(dispatchSpy, 2);
     sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
       errorHandlerId: root.instance().props.errorHandler.id,
       username,
     }));
+    sinon.assert.calledWith(dispatchSpy, fetchUserNotifications({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
   });
 
-  it('does not dispatch fetchUserAccount action if there is no username', () => {
-    const { store } = signInUserWithUsername('tofumatt');
+  it('only dispatches fetchUserNotifications action when the current logged-in user is being edited', () => {
+    // We do not have access to the user notifications for the current
+    // logged-in user because this user is loaded in Redux when authenticated,
+    // and we do not automatically load the notifications.
+
+    const username = 'tofumatt';
+
+    const { store } = signInUserWithUsername(username);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    // This happens when loading the user edit profile page of the current
+    // logged-in user (e.g., page refresh).
+    renderUserProfileEdit({ errorHandler, params: {}, store });
+
+    sinon.assert.calledOnce(dispatchSpy);
+    sinon.assert.calledWith(dispatchSpy, fetchUserNotifications({
+      errorHandlerId: errorHandler.id,
+      username,
+    }));
+  });
+
+  it('does not dispatch any actions if the current logged-in user is being edited and the notifications are loaded', () => {
+    const username = 'tofumatt';
+
+    const { store } = signInUserWithUsername(username);
     const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    store.dispatch(loadUserNotifications({
+      username,
+      notifications: createUserNotificationsResponse(),
+    }));
+
+    dispatchSpy.reset();
 
     // This happens when loading the user edit profile page of the current
     // logged-in user (e.g., page refresh).
-    renderUserProfileEdit({ params: {}, store });
+    renderUserProfileEdit({ errorHandler, params: {}, store });
```

Comment:
Why did you need to create an `errorHandler` in this test now, when one wasn't needed before? We're not asserting anything about it.

Reply:
copy/pasta 😞